pkg/loadbalancer/proxy.go

```diff
@@ -50,14 +51,37 @@ static_resources:
   {{- range $index, $data := .ServicePorts }}
   - name: listener_{{$index}}
     address:
-      socket_address: { address: {{ $data.Listener.Address }}, port_value: {{ $data.Listener.Port }} }
+      socket_address:
+        address: {{ $data.Listener.Address }}
+        port_value: {{ $data.Listener.Port }}
+        protocol: {{ $data.Listener.Protocol }}
+    {{- if eq $data.Listener.Protocol "UDP"}}
+    udp_listener_config:
+      downstream_socket_config:
+        max_rx_datagram_size: 9000
+    listener_filters:
+    - name: envoy.filters.udp_listener.udp_proxy
+      typed_config:
+        '@type': type.googleapis.com/envoy.extensions.filters.udp.udp_proxy.v3.UdpProxyConfig
+        stat_prefix: cluster_{{$index}}
+        matcher:
+          on_no_match:
+            action:
+              name: route
+              typed_config:
+                '@type': type.googleapis.com/envoy.extensions.filters.udp.udp_proxy.v3.Route
+                cluster: cluster_{{$index}}
+        upstream_socket_config:
+          max_rx_datagram_size: 9000
```

Comment:
it seems weird that the UDP section is so different from the TCP section but I assume it would make sense if I knew more about envoy config...

Reply:
It seems there are some internal details that make it different https://www.envoyproxy.io/docs/envoy/latest/configuration/listeners/udp_filters/udp_proxy , is interesting how complex must be to support UDP in the load balancing worlds, we can see haproxy does not want to implement it